src/orquestra/sdk/_base/_in_process_runtime.py

```
@@ -125,12 +143,31 @@ def get_workflow_run_outputs_non_blocking(
     ) -> t.Sequence[t.Any]:
         return self.get_workflow_run_outputs(workflow_run_id)
 
-    def get_available_outputs(self, workflow_run_id: WfRunId) -> t.Dict[str, t.Any]:
-        workflow_run_outputs = self.get_workflow_run_outputs(workflow_run_id)
-        output_task_ids = self._workflow_def_store[workflow_run_id].output_ids
-        return dict(zip(output_task_ids, workflow_run_outputs))
+    def get_available_outputs(
+        self, workflow_run_id: WfRunId
+    ) -> t.Dict[ir.TaskInvocationId, TaskOutputs]:
+        wf_def = self._workflow_def_store[workflow_run_id]
+
+        inv_outputs: t.Dict[ir.TaskInvocationId, TaskOutputs] = {}
+        for inv in wf_def.task_invocations.values():
+            output_vals = tuple(
+                [
+                    self._artifact_store[workflow_run_id][art_id]
+                    for art_id in inv.output_ids
+                ]
+            )
+
+            if len(inv.output_ids) == 1:
+                # This was a single-output task
+                output_vals = output_vals[0]
 
-    def get_full_logs(self, _) -> t.Dict[str, t.List[str]]:
+            inv_outputs[inv.id] = output_vals
+
+        return inv_outputs
+
+    def get_full_logs(
+        self, run_id: t.Union[WorkflowRunId, TaskRunId, None] = None
+    ) -> t.Dict[str, t.List[str]]:
         return {}
```

suggestion: maybe throw an exception here. Its better than silently return always nothing

---

Added!